libs/wire-api/src/Wire/API/Routes/Public/Brig.hs

```diff
@@ -535,33 +536,56 @@ type UserClientAPI =
                :> "prekeys"
                :> Get '[JSON] [PrekeyId]
            )
-    :<|> NonceAPI
-
-type NonceAPI =
-  -- be aware that the order matters, if get was first, then head requests would be routed to the get handler
-  NewNonce "head-nonce" 'HEAD 200
+    -- be aware that the order of the head-nonce and get-nonce matters, if get was first, then head requests would be routed to the get handler
+    :<|> NewNonce "head-nonce" 'HEAD 200
     :<|> NewNonce "get-nonce" 'GET 204
+    :<|> Named
+           "create-access-token"
+           ( Summary "Create a JWT DPoP access token"
+               :> Description
+                    ( "Create an JWT DPoP access token for the client CSR, given a JWT DPoP proof, specified in the `DPoP` header. \
+                      \The access token will be returned as JWT DPoP token in the `DPoP` header."
+                    )
+               :> QualifiedCaptureUserId "uid"
+               :> "clients"
+               :> CaptureClientId "client"
+               :> "access-token"
+               :> Header "DPoP" Proof
+               :> MultiVerb1
+                    'POST
+                    '[JSON]
+                    ( WithHeaders
+                        '[Header "Cache-Control" CacheControl]
+                        (DPoPAccessTokenResponse, CacheControl)
+                        (Respond 200 "Access token created" DPoPAccessTokenResponse)
+                    )
+           )
 
 type NewNonce name method statusCode =
   Named
     name
-    ( Summary "Get a new nonce for a client CSR, specified in the response header `Replay-Nonce` as a uuidv4 in base64url encoding"
+    ( Summary "Get a new nonce for a client CSR"
+        :> Description "Get a new nonce for a client CSR, specified in the response header `Replay-Nonce` as a uuidv4 in base64url encoding."
         :> ZUser
         :> "nonce"
         :> "clients"
         :> MultiVerb1
```

Comment:
```suggestion
        :> Capture "cid" ClientId  -- (unless there is a dedicated combinator for that like for uid above?)
        :> MultiVerb1
```

this allows us to pass the `ClientId` to the `newNonce` function that needs it as of https://github.com/wireapp/wire-server/pull/2655